### PR TITLE
Replace BatchNorm with weight norm in location and binary encoders

### PR DESCRIPTION
**Pull request by dereklhansen**

### CODE REVIEW

**Comment by dereklhansen** [comment]
- [ ] Pass tests with GPU on M2 
- [ ] Retrain binary and sleep on sdss_galaxies
- [ ] Retrain binary and sleep on sdss_galaxies_vae

**Comment by dereklhansen** [comment]
So far, just replacing BatchNorm with weight norm everywhere leads to very poor performance on the M2 example

**Comment by ismael-mendoza** [comment]
That's unfortunate, maybe we do need batchnorm after all :( 

Maybe you have other ideas to improve weight norm but if we do have to stick with batchnorm we could create a custom version that doesn't trip us up

**Comment by dereklhansen** [comment]
> That's unfortunate, maybe we do need batchnorm after all :(
> 
> Maybe you have other ideas to improve weight norm but if we do have to stick with batchnorm we could create a custom version that doesn't trip us up

It turns out batch norm wasn't the problem with the different output depending on image size. So I think it's fine to leave it for now.
